src/WordsMove.java

```diff
@@ -59,7 +59,13 @@ public void doExecute(WordsObject object, WordsEnvironment environment) {
 	@Override
 	public LinkedList<WordsAction> doExpand(WordsObject object) {
 		if (distanceExpression != null) {
-			AST.ASTValue value = distanceExpression.eval(null);
+			AST.ASTValue value;
+			try {
+				value = distanceExpression.eval(null);
```

Comment:
Putting null here (for the environment) won't work, because the expression being evaluated may include object property references, so it'll need access to the environment.  Which means the doExpand and expand methods will need the environment too.